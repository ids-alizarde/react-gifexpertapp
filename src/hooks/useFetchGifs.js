import { useState, useEffect } from 'react';
import { getGifs } from '../services/GetGifs';  

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        getGifs( category )
            .then( images => {
                setState({
                    data: images,
                    loading: false
                });
            });
    }, [ category ]);

    return state;

}
