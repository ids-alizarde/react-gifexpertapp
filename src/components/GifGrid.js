import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../services/GetGifs';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ]);

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeInTopLeft">{ category }</h3>

            { loading && <p>Loading.........</p> }

            <div className="card-grid">

                { images.map( img => (

                    <GifGridItem 
                        key={img.id}
                        { ...img }>

                    </GifGridItem>

                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
