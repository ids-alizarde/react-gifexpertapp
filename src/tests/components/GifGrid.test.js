import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {

    const category = 'Charly y los numeros';

    test('Debe mostrar el componente <GifGrid/> correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        }); 

        const wrapper = shallow( <GifGrid category = { category } /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Titulo X'
        }, 
        {
            id: '123',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Titulo X'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        }); 

        const wrapper = shallow( <GifGrid category = { category } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'p' ).exists() ).toBe( false );
        expect( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length );

    });
    
});
