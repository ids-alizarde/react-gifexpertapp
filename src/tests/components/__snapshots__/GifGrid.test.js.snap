// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInTopLeft"
  >
    Charly y los numeros
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo X"
      url="https://localhost/cualquiercosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Titulo X"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe mostrar el componente <GifGrid/> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInTopLeft"
  >
    Charly y los numeros
  </h3>
  <p>
    Loading.........
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
